### PR TITLE
AB#3080 -- swtich userOrigin cookie to session variable

### PR DESCRIPTION
**Pull request by leiye29**
### Description

1. remove userOrigin cookie
2. add userOrigin as session variable to handle dynamic MSCA/MSCA-D link

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/root.tsx [hunk 1]
```diff
@@ -7,8 +7,9 @@ import { Links, LiveReload, Meta, Outlet, Scripts, ScrollRestoration, useLoaderD
 import { useTranslation } from 'react-i18next';
 import { getToast } from 'remix-toast';
 
+import { getSessionService } from './services/session-service.server';
```
Comment (gregory-j-baker) [inline frontend/app/root.tsx:10]:
```suggestion
import { getSessionService } from '~/services/session-service.server';
```

File: frontend/app/root.tsx [hunk 2]
```diff
@@ -7,8 +7,9 @@ import { Links, LiveReload, Meta, Outlet, Scripts, ScrollRestoration, useLoaderD
...
 import { getFixedT, getLocale } from './utils/locale-utils.server';
```
Comment (gregory-j-baker) [inline frontend/app/root.tsx:11]: I know this isn't a part of your change, but can you modify this line, too?

```suggestion
import { getFixedT, getLocale } from '~/utils/locale-utils.server';
```

File: frontend/app/root.tsx [hunk 3]
```diff
@@ -7,8 +7,9 @@ import { Links, LiveReload, Meta, Outlet, Scripts, ScrollRestoration, useLoaderD
...
-import { getUserOrigin, userOriginCookie } from './utils/user-origin-utils.server';
+import { getUserOrigin } from './utils/user-origin-utils.server';
```
Comment (gregory-j-baker) [inline frontend/app/root.tsx:12]:
```suggestion
import { getUserOrigin } from '~/utils/user-origin-utils.server';
```

**Review by gregory-j-baker — APPROVED** [review]
Approved with trivial changes requested (sorry).

File: frontend/app/root.tsx (hunk 1 above)
Comment (leiye29) [reply frontend/app/root.tsx]: modified

File: frontend/app/root.tsx (hunk 2 above)
Comment (leiye29) [reply frontend/app/root.tsx]: modified

File: frontend/app/root.tsx (hunk 1 above)
Comment (leiye29) [reply frontend/app/root.tsx]: modified

File: frontend/app/root.tsx (hunk 3 above)
Comment (leiye29) [reply frontend/app/root.tsx]: modified

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**